### ISSUE TITLE
"%<width>ES" in `formatTime` doesn't allow specifying width of decimal places

### ISSUE THREAD

**Issue by jumper149** [haskell/time#120]
I feel like this is a problem with with the "ES" specifier. I am using time-1.9.3.

```haskell
λ *Main Data.Time > t = Data.Maybe.fromMaybe undefined $ parseTimeM False defaultTimeLocale "%ES" "1.245678" :: DiffTime 
λ *Main Data.Time > formatTime defaultTimeLocale "%3ES" t
"01.245678000000"
λ *Main Data.Time > formatTime defaultTimeLocale "%ES" t
"1.245678"
```

The expected would behaviour would be to show 3 decimal places instead of the default 12.

**Comment by AshleyYakeley** [comment]
Defect confirmed in both `%3ES` and `%3Es`.

**Comment by AshleyYakeley** [comment]
Fixed for 1.10.
